tensorflow/python/keras/layers/preprocessing/preprocessing_stage.py

```diff
@@ -94,3 +96,120 @@ def map_fn(x):
                                              reset_state=reset_state)
 
 
+class FunctionalPreprocessingStage(base_preprocessing_layer.PreprocessingLayer,
+                                   functional.Functional):
+  """A functional preprocessing stage.
+
+  This preprocessing stage wraps a graph of preprocessing layers into a
+  Functional-like object that enables you to `adapt()` the whole graph via
+  a single `adapt()` call on the preprocessing stage.
+
+  Arguments:
+    inputs: List of input tensors (must be created via `tf.keras.Input()`).
+    outputs: List of outputs tensors.
+    name: String, optional. Name of the model.
+    trainable: Boolean, whether the model's variables should be trainable.
```

Comment:
Is there any case where a FunctionalPreprocessingStage is trainable?

Reply:
I am slightly uncertain about this. From the test case of (sequential) PreprocessingStage it seems like we are allow mixing preprocessing layers with regular layers in stages. In that case stages may be trainable? And FunctionalPreprocessingStage is trying to mirror the sequential one. @fchollet what do you think

Reply:
Yes, that's correct, they may be trainable (like any other layer). However it would be conceptually wrong to have a trainable layer before an adaptable layer. But you could mix trainable layers and non-adaptable preprocessing layers, for instance.

Please remove this arg, as we don't need a constructor arg and can simply rely on the `self.trainable` property.

Reply:
Thanks for the clarification. The arg is removed from docstring.